aca_required_temp.py

```diff
@@ -0,0 +1,189 @@
+                                        cold_t_ccd=COLD_T_CCD,
+                                        warm_t_ccd=WARM_T_CCD)
+        TEMP_CACHE[id_hash] = (t_ccd, n_acq)
+    return t_ccd
+
+
+def get_rolldev(pitch):
+    if pitch < ROLL_TABLE['pitch'][0]:
+        return 0
+    idx = np.searchsorted(ROLL_TABLE['pitch'], pitch)
+    return ROLL_TABLE['rolldev'][idx - 1]
+
+
+def best_temp_roll(ra, dec, nom_roll, day_pitch, time, cone_stars):
+    """
+    Loop over possible roll range for this pitch and return best temp/roll
+    combination
+    """
+    best_t_ccd = None
+    best_roll = None
+    rolldev = get_rolldev(day_pitch)
+    # Quat doesn't care about the domain of roll, so I don't have to
+    # do anything tricky with 180/360 etc
+    for rolldiff in np.arange(0, rolldev, step=.5):
+        # Walk out from nom_roll instead of just through the range,
+        # and quit if we get one that is at the warm limit
+        # That will give us the closest-to-nominal best choice
+        for roll in [nom_roll - rolldiff, nom_roll + rolldiff]:
+            roll_t_ccd = max_temp(ra, dec, roll, time=time, cone_stars=cone_stars)
+            if roll_t_ccd is not None:
+                if best_t_ccd is None or roll_t_ccd > best_t_ccd:
+                    best_t_ccd = roll_t_ccd
+                    best_roll = roll
+                if best_t_ccd == WARM_T_CCD:
+                    break
+    return best_t_ccd, best_roll
+
+
+def temps_for_attitude(ra, dec, start='2014-09-01', stop='2015-12-31'):
+    # reset the caches at every new attitude
+    global TEMP_CACHE
+    TEMP_CACHE.clear()
+
+    # set the agasc lookup time to be in the middle of the cycle for
+    # proper motion correction
+    agasc_mid_time = DateTime((DateTime(start).secs + DateTime(stop).secs)
+                              / 2).date
+
+    # Get stars in this field
+    cone_stars = get_agasc_cone(ra, dec, time=agasc_mid_time)
+
+    # get a list of days
+    start = DateTime(start)
+    stop = DateTime(stop)
+    days = start + np.arange(stop - start)
+
+    temps = {}
+    # loop over them
+    for day in days.date:
+        nom_roll = nominal_roll(ra, dec, time=day)
+        day_pitch = pitch(ra, dec, time=day)
+        if day_pitch < 45 or day_pitch > 170:
+            continue
+
+        nom_roll_t_ccd = max_temp(ra, dec, nom_roll, time=day, cone_stars=cone_stars)
+        # if we can get the nominal roll catalog at warmest temp, why check the rest?
+        if nom_roll_t_ccd == WARM_T_CCD:
+            best_t_ccd = nom_roll_t_ccd
+            best_roll = nom_roll
+        else:
+            best_t_ccd, best_roll = best_temp_roll(ra, dec, nom_roll,
+                                                   day_pitch,
+                                                   time=day,
+                                                   cone_stars=cone_stars)
+        # should we have values or skip entries for None here?
```

Comment:
The worst possible field would be one that has no stars brighter than 11.0.  I actually don't think that exists, but even if it does then a table with values that essentially say "can't observe this" will be generated.

This reminds me that we probably also want fields `nom_n_acq` and `best_n_acq` that provide the expected mean number of acq stars acquired.